lib/internals.js

```diff
@@ -902,6 +908,42 @@ module.exports = (self) => {
       return mergeConfigs;
     },
 
+    /**
+     * Gets postcss config for the current environment *
+     *
+     * @param {object} buildOptions: build options
+     * @param {string} id: apos / public
+     *
+     * @returns {Promise<import('vite').UserConfig>}
+     */
+    async getPostcssConfig(buildOptions, id) {
+      const {
+        enable: enablePostcssViewport, ...postcssViewportOptions
+      } = buildOptions?.postcssViewportToContainerToggle || {};
+
+      const postcssPlugins = [];
+      if (id === 'public') {
+        try {
+          const {
+            plugins
+          } = await postcssrc({}, self.apos.rootDir);
+          postcssPlugins.push(...plugins);
+        } catch (err) { /* Project has no postcss config file */ }
+      }
+
+      if (enablePostcssViewport) {
+        postcssPlugins.push(postcssViewportToContainerToggle(postcssViewportOptions));
+      }
+
+      if (id === 'apos') {
+        postcssPlugins.push(
+          require('autoprefixer')()
+        );
+      }
+
+      return vitePostcssConfig({ plugins: postcssPlugins });
+    },
```

Comment:
Moved the logic here, tried the respect the plugins order. options coming from core.